### PR TITLE
Clean up CompressBlock/CompressBlockInternal a bit

### PR DESCRIPTION
**Pull request by ltamasi**
Summary:
The patch cleans up and refactors `CompressBlock` and `CompressBlockInternal` a bit.
In particular, it does the following:
* It renames `CompressBlockInternal` to `CompressData` and moves it to `util/compression.h`,
where other general compression-related utilities are located. This will facilitate reuse in the
BlobDB write path.
* The signature of the method is changed so it now takes `compression_format_version`
(similarly to the compression library specific methods) instead of `format_version` (which is
specific to the block based table).
* `GetCompressionFormatForVersion` no longer takes `compression_type` as a parameter.
This parameter was only used in a (not entirely up-to-date) assertion; also, removing it
eliminates the need to ensure this precondition holds at all call sites.
* Does some minor cleanup in `CompressBlock`, for instance, it is now possible to pass
only one of `sampled_output_fast` and `sampled_output_slow`.

Test Plan:
`make check`

### CODE REVIEW

File: table/format.h [hunk 1]
```diff
@@ -110,15 +110,7 @@ struct IndexValue {
...
-inline uint32_t GetCompressFormatForVersion(CompressionType compression_type,
-                                            uint32_t version) {
-#ifdef NDEBUG
-  (void)compression_type;
-#endif
-  // snappy is not versioned
-  assert(compression_type != kSnappyCompression &&
-         compression_type != kXpressCompression &&
-         compression_type != kNoCompression);
+inline uint32_t GetCompressFormatForVersion(uint32_t version) {
```
[the remark this answers is not in the document]
Comment (riversand963) [reply table/format.h:113]: iiuc, `compress_format` is determined only by `format_version`. The original assertions can be done outside this fuction. Is this correct?

File: table/block_based/block_based_table_builder.cc [hunk 2]
```diff
@@ -170,22 +113,19 @@ Slice CompressBlock(const Slice& raw, const CompressionInfo& info,
...
 
   // If requested, we sample one in every N block with a
   // fast and slow compression algorithm and report the stats.
   // The users can use these stats to decide if it is worthwhile
   // enabling compression and they also get a hint about which
   // compression algorithm wil be beneficial.
   if (do_sample && info.SampleForCompression() &&
-      Random::GetTLSInstance()->OneIn((int)info.SampleForCompression()) &&
-      sampled_output_fast && sampled_output_slow) {
+      Random::GetTLSInstance()->OneIn((int)info.SampleForCompression())) {
```
[the remark this answers is not in the document]
Comment (ltamasi) [reply table/block_based/block_based_table_builder.cc:126]: Good call, thanks!

File: table/format.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (ltamasi) [reply table/format.h:113]: Yes, `compress_format_version` is (and has always been) a function of only `format_version`.

The idea behind this assertion was that there are only four compression types (zlib, bz2, and the two lz4 variants) where `compress_format_version` has an effect; the assertion's goal was to ensure that the method is not even called for other types. Then again, the list has not been kept up to date, and I actually think that's it's simpler/cleaner just to call `GetCompressFormatForVersion` for all compression types than to put `if` conditions at all call sites to make sure the assertion does not fire. (Also, the check in this method is super cheap.)

[the remark this answers is not in the document]
Comment (riversand963) [reply table/format.h:113]: Just a thought, I do not have a strong opinion on this.
maybe the check still has a value: make sure this function is called only for certain compression types that are actually meaningful.

**Pull request closed without merging**